includes/blocks/class-sensei-course-categories-block.php

```diff
@@ -69,24 +59,39 @@ public function render_block( $attributes, $content, WP_Block $block ): string {
 			return '';
 		}
 
-		$css             = Sensei_Block_Helpers::build_styles( $attributes );
-		$wrapper_classes = 'taxonomy-course-category';
+		$wrapper_classes = [ 'taxonomy-course-category' ];
 
 		if ( isset( $attributes['textAlign'] ) ) {
-			$wrapper_classes .= ' has-text-align-' . $attributes['textAlign'];
+			$wrapper_classes[] = 'has-text-align-' . $attributes['textAlign'];
 		}
 
-		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $wrapper_classes ) );
-		$link_attributes    = '<a ' . Sensei_Block_Helpers::render_style_attributes( $wrapper_attributes, $css );
+		list( $wrapper_style ) = Sensei_Block_Helpers::css_variables(
```

Comment:
I wonder if it would make sense to keep all the presentation part in the frontend, and replace only the dynamic part here. Wouldn't it make it simpler?

Reply:
@renatho Could you please provide more details? I didn't follow the suggestion.

Reply:
Hi @gabrielcaires!

I mean to use the `save` from the block registration, containing the needed classes and styles (reusing from the `edit`?). And then use this as the replacement (this would come in the `$content` of this method). But please, take into account that I didn't dig into that, so maybe there's a strong reason for this approach that I didn't realize. 😉

Reply:
@renatho sorry for the delay.
I liked your suggestion, so I moved every style and class logic to be done directly on the editor and the PHP size is only getting it and wrapping the list of terms.


My favorites pros are:
- Avoid implementing the rendering part two times, so every time that a new attribute is added is not necessary to update the PHP file. 
-  I could write tests to cover the saving method, giving me more control over it. 
- I  tested it enabling more features using the Gutenberg support and all of them are working fine with no changes


The main point that I haven't a success was removing these lines: 

https://github.com/Automattic/sensei/pull/5610/files#diff-ca728889b5f81916884b66ff250e343f50821f49be6ae2730d2b7dabedf2c45cR61-R68


Unfortunately, the save method doesn't receive the colors from the `withColors`  hook, so I had to save the colors in order to get them on the save method.

Reply:
Nice! Thank you for that!
I just couldn't see what's in the link you sent because it didn't work.

Reply:
Oh Sorry, link update here:
https://github.com/Automattic/sensei/pull/5610/files#diff-ca728889b5f81916884b66ff250e343f50821f49be6ae2730d2b7dabedf2c45cR52-R64

I mentioned the useEffects callbacks we are using to update the block attributes.